client/gutenberg/extensions/publicize/connection.jsx

```diff
@@ -56,6 +69,32 @@ class PublicizeConnection extends Component {
 			</Notice>
 		);
 
+	/**
+	 * Displays a message when a connection requires reauthentication. We used this when migrating LinkedIn API usage from v1 to v2,
+	 * since the prevous OAuth1 tokens were incompatible with OAuth2.
+	 *
+	 * @returns {object|null} Notice about reauthentication
+	 */
+	maybeDisplayLinkedInNotice = () =>
+		this.connectionNeedsReauth() && (
+			<Notice className="jetpack-publicize-notice" isDismissible={ false } status="error">
+				<p>
+					{ __(
+						'Your LinkedIn connection needs to be reauthenticated to continue working – head to Sharing to take care of it.'
+					) }
+				</p>
+				<ExternalLink href={ getSharingLink() }>{ __( 'Go to Sharing settings' ) }</ExternalLink>
+			</Notice>
+		);
+
+	/**
+	 * Check whether the connection needs to be reauthenticated.
+	 *
+	 * @returns {boolean} True if connection must be reauthenticated.
+	 */
+	connectionNeedsReauth = () =>
+		this.props.mustReauthConnections.some( connection => connection === this.props.name );
```

Comment:
I'd prefer the safer lodash `some` here
```suggestion
		some( this.props.mustReauthConnections, connection => connection === this.props.name );
```

`withSelect` will return `undefined` until the resolvers supply the data.

```js
undefined.some()   // 💥
some( undefined )  // 👌
```

Reply:
Changed to lodash's `some`